ls1intum/Artemis
# `General`: Fix an issue when deleting users which are referenced in notifications

### Checklist
#### General


#### Server
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).#### Client

### Motivation and Context
See #6102: Can not delete users because of connected notifications.

### Description
Delete notifications of user before deleting users.

### Steps for Testing
1. Create an user
2. Generate a notification for that user (e.g. by replying to an exercise post of that user)
3. Delete the user
4. There should be no error
